tests/cpp/test_host_irs.cpp

```diff
@@ -387,6 +389,145 @@ INSTANTIATE_TEST_SUITE_P(
+  auto set_stream = IrBuilder::create<PostOnStream>(
+      static_cast<IrContainer*>(hic.get()),
+      stream,
+      std::vector<Val*>(),
+      std::vector<Val*>());
+  hic->pushBackTopLevelExprs(set_stream);
+
+  HostIrExecutor hie(std::move(hic));
+  setCurrentCUDAStream(c10::cuda::getDefaultCUDAStream(0));
+  hie.runWithInput({});
+  EXPECT_NE(
+      c10::cuda::getDefaultCUDAStream(0), c10::cuda::getCurrentCUDAStream(0));
+}
+
+using StreamHostIrTestParams = std::tuple<bool, int, int>;
+using StreamHostIrTest = NVFuserFixtureParamTest<StreamHostIrTestParams>;
+
+// The following test execute the same fusion `n_iterations` times by posting
+// the kernels on `n_streams` different streams in a Round-Robin fashion. We
+// thus produce `n_iterations` outputs from the same input, with a potential
+// overlap of n_streams/n_iterations
+TEST_P(StreamHostIrTest, SingleFusionMultipleStreams) {
+  auto [use_fusion_executor_cache, n_streams, n_iterations] = GetParam();
+
+  // [Step 1)] Define the Fusion we want to execute
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  std::vector<int64_t> input_sizes = {4, 8, 32};
+
+  auto tv0 = makeConcreteTensor(input_sizes);
+  auto tv1 = add(tv0, tv0);
+  auto tv2 = sum(tv1, {0});
+  fusion->addInput(tv0);
+  fusion->addOutput(tv2);
+
+  // [Step 2)] Instantiate an HostIroCntainer
+  auto hic = std::make_unique<HostIrContainer>();
+  FusionGuard::setCurFusion(hic.get());
+
+  // Create N different Streams
+  std::vector<Stream*> streams;
+  for (int i = 0; i < n_streams; i++) {
+    streams.push_back(
+        IrBuilder::create<Stream>(static_cast<IrContainer*>(hic.get())));
+  }
+
+  // [Step 3)] Create a HostUnit Ir holding the created fusion
+  auto host_unit = IrBuilder::create<HostUnit>(
+      static_cast<IrContainer*>(hic.get()), std::move(fusion));
+
+  // [Step 4)] Create TensorViews representing the Fusion's inputs at the Host
+  // level
+  IrCloner ir_cloner_input(hic.get());
+  std::vector<Val*> post_on_stream_inputs = {
+      ir_cloner_input.clone(host_unit->fusion_to_execute()->inputs().at(0))};
+  hic->addInput(post_on_stream_inputs.at(0));
+
+  for (int i = 0; i < n_iterations; i++) {
+    // [Step 4)] Create TensorViews representing the Fusion's ouputs at the Host
+    // level
+    IrCloner ir_cloner_output(hic.get());
+    std::vector<Val*> post_on_stream_outputs = {ir_cloner_output.clone(
+        host_unit->fusion_to_execute()->outputs().at(0))};
+
+    // [Step 5)] Create a PostOnStream Ir representing executing the Fusion with
+    // given I/O
+    auto post_on_stream = IrBuilder::create<PostOnStream>(
+        static_cast<IrContainer*>(hic.get()),
+        host_unit,
+        post_on_stream_inputs,
+        post_on_stream_outputs);
+
+    // Set the Stream
+    auto set_stream = IrBuilder::create<PostOnStream>(
+        static_cast<IrContainer*>(hic.get()),
+        streams[i % streams.size()],
+        std::vector<Val*>(),
```

Comment:
```suggestion
        /*inputs=*/std::vector<Val*>(),
```

Reply:
https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments

Reply:
not relevant anymore since we use `SetCurrentStream` IR which takes no I/O arguments